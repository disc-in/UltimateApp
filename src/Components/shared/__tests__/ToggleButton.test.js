import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import ToggleButton from '../ToggleButton';

describe('<ToggleButton />', () => {
  const onValueChange = jest.fn();
  const value = 'animation';
  const icons = ['clipboard-outline', 'video'];
  const possibleValues = ['animation', 'video'];

  it('renders correctly', () => {
    const { toJSON } = render(
      <ToggleButton value={value} onValueChange={onValueChange} icons={icons} possibleValues={possibleValues} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('delegates onValueChange prop', async () => {
    const { getByTestId } = render(
      <ToggleButton value={value} onValueChange={onValueChange} icons={icons} possibleValues={possibleValues} />,
    );

    await fireEvent.press(getByTestId('toggle-video'));

    expect(onValueChange).toHaveBeenCalledWith('video');
  });
});
