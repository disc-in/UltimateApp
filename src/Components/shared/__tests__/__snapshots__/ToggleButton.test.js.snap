// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "center",
        "marginTop": 5,
      },
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
        "selected": true,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "height": 36,
            "width": 36,
          },
          Object {
            "opacity": 0.32,
          },
          Array [
            Object {
              "height": 42,
              "margin": 0,
              "width": 42,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, .08)",
              "borderColor": "rgba(0, 0, 0, 0.29)",
              "borderRadius": 4,
            },
            Array [
              Object {
                "borderWidth": 0.5,
              },
              Object {
                "borderBottomRightRadius": 0,
                "borderTopRightRadius": 0,
              },
              Array [
                Object {
                  "borderColor": "#A6A6A6",
                  "borderWidth": 0.5,
                },
                Object {
                  "borderBottomRightRadius": 0,
                  "borderTopRightRadius": 0,
                },
                Object {
                  "backgroundColor": "#e5e5e5",
                  "opacity": 0.8,
                },
              ],
            ],
          ],
        ],
      ]
    }
    testID="toggle-animation"
  >
    <View>
      <Text />
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
        "selected": false,
      }
    }
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "height": 36,
            "width": 36,
          },
          false,
          Array [
            Object {
              "height": 42,
              "margin": 0,
              "width": 42,
            },
            Object {
              "backgroundColor": "transparent",
              "borderColor": "rgba(0, 0, 0, 0.29)",
              "borderRadius": 4,
            },
            Array [
              Object {
                "borderWidth": 0.5,
              },
              Object {
                "borderBottomLeftRadius": 0,
                "borderLeftWidth": 0,
                "borderTopLeftRadius": 0,
              },
              Array [
                Object {
                  "borderColor": "#A6A6A6",
                  "borderWidth": 0.5,
                },
                Object {
                  "borderBottomLeftRadius": 0,
                  "borderLeftWidth": 0,
                  "borderTopLeftRadius": 0,
                },
                undefined,
              ],
            ],
          ],
        ],
      ]
    }
    testID="toggle-video"
  >
    <View>
      <Text />
    </View>
  </View>
</View>
`;
